### PR TITLE
[NNVM/TOPI][OP] Split : default axis to 0 and allow negative values - nump…

### PR DESCRIPTION
**Pull request by srkreddy1238**
…y consistency. Use cpp implementation for available ops.

Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from others in the community.

### CODE REVIEW

**Comment by srkreddy1238** [comment]
@siju-samuel @PariksheetPinjari909 @yzhliu & @nishi-t welcome to review.

**Comment by PariksheetPinjari909** [comment]
Please add test cases for your modification

File: nnvm/src/top/tensor/transform.cc [hunk 1]
```diff
@@ -344,34 +344,41 @@ inline bool SplitInferShape(const NodeAttrs& attrs,
...
   if (dshape.ndim() == 0) return false;
 
+  auto axis = param.axis;
+  if (axis < 0) {
+    axis += dshape.ndim();
+  }
+  CHECK_LT(axis, dshape.ndim())
+    << "axis should be with in input dimension range";
+  CHECK_GT(axis, -1)
+    << "axis should be with in input dimension range " << axis;
```
Comment (nishi-t) [inline nnvm/src/top/tensor/transform.cc]: I'd suggest clarifying the value to be displayed.

for example:
```
"axis should be within input dimension range but got " <<  axis
```

File: nnvm/src/top/tensor/transform.cc [hunk 2]
```diff
@@ -344,34 +344,41 @@ inline bool SplitInferShape(const NodeAttrs& attrs,
   const TShape& dshape = (*in_shape)[0];
   if (dshape.ndim() == 0) return false;
 
+  auto axis = param.axis;
+  if (axis < 0) {
+    axis += dshape.ndim();
+  }
+  CHECK_LT(axis, dshape.ndim())
+    << "axis should be with in input dimension range";
```
Comment (nishi-t) [inline nnvm/src/top/tensor/transform.cc]: It will be helpful to display a value of the axis as with following

**Comment by tqchen** [comment]
Let us keep the convention of NNVM as it is, to allow backward compatibility. The full numpy consistency change need to will happen in relay

**Comment by srkreddy1238** [comment]
@tqchen retained backward compatibility.

**Pull request merged**